cordyceps/src/mpsc_queue.rs

```diff
@@ -1283,6 +1395,99 @@ mod tests {
         }
     }
 
+    #[test]
+    fn basically_works_all_const() {
+        use std::thread;
+
+        const THREADS: i32 = if_miri(3, 8);
+        const MSGS: i32 = if_miri(10, 1000);
+
+        static STUB_ENTRY: Entry = const_stub_entry(666);
+        static MPSC: MpscQueue<Entry> =
+            unsafe { MpscQueue::<Entry>::new_with_static_stub(&STUB_ENTRY) };
+
+        assert_eq!(MPSC.dequeue(), None);
+
+        let threads: Vec<_> = (0..THREADS)
+            .map(|thread| {
+                thread::spawn(move || {
+                    for i in 0..MSGS {
+                        MPSC.enqueue(entry(i));
+                        println!("thread {}; msg {}/{}", thread, i, MSGS);
+                    }
+                })
+            })
+            .collect();
+
+        let mut i = 0;
+        while i < THREADS * MSGS {
+            match MPSC.try_dequeue() {
+                Ok(msg) => {
+                    i += 1;
+                    println!("recv {:?} ({}/{})", msg, i, THREADS * MSGS);
+                }
+                Err(TryDequeueError::Busy) => {
+                    panic!("the queue should never be busy, as there is only one consumer")
+                }
+                Err(e) => {
+                    println!("recv error {:?}", e);
+                    thread::yield_now();
+                }
+            }
+        }
+
+        for thread in threads {
+            thread.join().unwrap();
+        }
```

Comment:
it occurs to me that the main body of most of these tests could probably be factored out, but we can do that later...